framework/web/XmlResponseFormatter.php

```diff
@@ -52,61 +52,86 @@ class XmlResponseFormatter extends Component implements ResponseFormatterInterfa
      */
     public $useTraversableAsArray = true;
     /**
-     * @var bool if object tags should be added
+     * @var bool if object tags should be added (convert class to tag name)
      * @since 2.0.11
      */
     public $useObjectTags = true;
+    /**
+     * @var bool if true, converts object tags to lowercase, `$useObjectTags` must be enabled
+     * @since 2.0.43
+     */
+    public $objectTagToLowercase = false;
+
+    /**
+     * @var DOMDocument the XML document, serves as the root of the document tree
+     * @since 2.0.43
+     */
+    protected $dom;
 
 
     /**
      * Formats the specified response.
+     *
      * @param Response $response the response to be formatted.
      */
     public function format($response)
     {
-        $charset = $this->encoding === null ? $response->charset : $this->encoding;
+        if ($this->encoding === null) {
+            $this->encoding = $response->charset;
```

Comment:
Causes side effect in the method. Let's revert this part.

Reply:
@samdark reverted